test/server/cards/01_SOR/units/ISBAgent.spec.ts

```diff
@@ -0,0 +1,75 @@
+describe('ISB Agent', function() {
+    integration(function(contextRef) {
+        describe('ISB Agent\' ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['confiscate', 'waylay', 'isb-agent'],
+                        groundArena: ['atst'],
+                        spaceArena: ['cartel-spacer']
+                    },
+                    player2: {
+                        hand: ['disarm'],
+                        groundArena: ['wampa'],
+                        spaceArena: ['alliance-xwing']
+                    }
+                });
+            });
+
+            describe('when event is revealed', function() {
+                it('should deal 1 damage to a unit', function () {
+                    const { context } = contextRef;
+
+                    context.player1.clickCard(context.isbAgent);
+
+                    expect(context.player1).toBeAbleToSelectExactly([context.confiscate, context.waylay]);
+                    expect(context.player1).toHaveExactPromptButtons(['Choose no target']);
```

Comment:
there is a method for this, toHaveChooseNoTarget or something like that

Reply:
`toHaveChooseNoTarget` doesn't do exactly the same thing because this also checks that there are no other buttons.

Reply:
Could we add an additional parameter to it like `allowOtherButtons = true` or something? Though the other similar methods would also need to be extended

Reply:
I guess ideally `toHaveExactPromptButton` should be able to accept custom matchers (e.g. `chooseNoTargetButton`, `passAbilityButton`, etc.).

I changed it to:
```Typescript
expect(context.player1).toHaveChooseNoTargetButton();
expect(context.player1).not.toHavePassAbilityButton();
```
because it was quicker :)